criu/files.c

```diff
@@ -382,7 +382,13 @@ static int fill_fd_params(struct pid *owner_pid, int fd, int lfd,
 	p->fs_type	= fsbuf.f_type;
 	p->fd		= fd;
 	p->pos		= fdinfo.pos;
-	p->flags	= fdinfo.flags;
+	/*
+	 * The kernel artificially adds the O_CLOEXEC flag on the file pointer
+	 * flags by looking at the flags on the file descriptor (see kernel
+	 * code fs/proc/fd.c). FD_CLOEXEC is a file descriptor property, which
+	 * is saved in fd_flags.
+	 */
+	p->flags	= fdinfo.flags & ~O_CLOEXEC;
```

Comment:
when creating a checkpoint of the following process `crit` shows a different result.
```C
#include <stdio.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

int main()
{
	int fd1 = open("foo", O_CREAT | O_RDWR | O_CLOEXEC, 0644);
	int fd2 = open("bar", O_CREAT | O_RDWR, 0644);
	getchar();
	close(fd1);
	close(fd2);
	return 0;
}
```
with this change:
```
# crit show files.img | grep flags
               ...
                "flags": "O_RDWR | O_LARGEFILE", 
                "flags": "O_RDWR | O_LARGEFILE", 
```
without this change:
```
# crit show files.img | grep flags
               ...
                "flags": "O_RDWR | O_LARGEFILE | O_CLOEXEC",
                "flags": "O_RDWR | O_LARGEFILE", 
```